src/test_application/VehicleBuilder.h

```diff
@@ -156,12 +162,12 @@ template <std::size_t N>
 std::array<PartId, N> VehicleBuilder::create_parts()
 {
     std::array<PartId, N> out;
-    m_data.m_partIds.create(std::begin(out), std::end(out));
+    m_data->m_partIds.create(std::begin(out), std::end(out));
 
-    std::size_t const capacity = m_data.m_partIds.capacity();
-    m_data.m_partMachCount.resize(capacity);
-    m_data.m_partPrefabs.resize(capacity);
-    m_data.m_partTransforms.resize(capacity);
+    std::size_t const capacity = m_data->m_partIds.capacity();
+    m_data->m_partMachCount.resize(capacity);
+    m_data->m_partPrefabs.resize(capacity);
```

Comment:
Consider making a helper function for these repeated resize calls

Reply:
This is the only place these are called, adding a new function would just unnecessary lines and more complexity in this case.

Additionally, these parallel vectors are everywhere, and it's possible to put them into single allocations. I haven't yet finalized the best way to do this, but I have a few bad ideas.